tdrs-backend/tdpservice/data_files/admin/admin.py

```diff
@@ -23,11 +26,42 @@ def has_change_permission(self, request, obj=None):
         """Read only permissions."""
         return False
 
-
 @admin.register(DataFile)
 class DataFileAdmin(ReadOnlyAdminMixin, admin.ModelAdmin):
     """Admin class for DataFile models."""
 
+    class Media:
+        """Media class for DataFileAdmin."""
+
+        js = ('admin/js/admin/admin_datafile_model.js',)
+
+    actions = ['reparse']
+
+    def reparse(self, request, queryset):
+        """Reparse the selected data files."""
+        files = queryset.values_list("id", flat=True)
+        file_ids = ",".join(map(str, files))
+        call_command("clean_and_reparse", f"-f {file_ids}")
+        self.message_user(
+            request,
+            ngettext(
+                "%d file successfully submitted for reparsing.",
+                "%d files successfully submitted for reparsing.",
+                files.count(),
+            )
+            % files.count(),
+            messages.SUCCESS,
+        )
+        return redirect("/admin/search_indexes/reparsemeta/")
+
+    def get_actions(self, request):
+        """Return the actions."""
+        actions = super().get_actions(request)
+        if not request.user.groups.filter(name__in=["OFA System Admin", "OFA Admin"]).exists():
+            if "reparse" in actions:
+                del actions["reparse"]
```

Comment:
Instead of another `if` statement can we just make it `actions.pop("reparse", None)`?

Reply:
Unfortunately no since actions is a list and .pop only works with int indexes

Reply:
If `actions` is a list how are you indexing into it with a string?

Reply:
you are right, it is a dictionary and although my original test failed, I tested again and it passed. so will do the change